lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/persistence/DBOStorageLocation.java

```diff
@@ -208,6 +209,18 @@ public String toString() {
 			
 			@Override
 			public DBOStorageLocation createDatabaseObjectFromBackup(DBOStorageLocation backup) {
+				// PLFM-5769 - Remove after migrated
+				if (backup.getData() != null && backup.getData() instanceof ExternalS3StorageLocationSetting) {
+					ExternalS3StorageLocationSetting data = (ExternalS3StorageLocationSetting) backup.getData();
+					if (data.getBaseKey() != null) {
+						// Base keys should not have trailing slashes
+						data.setBaseKey(removeTrailingSlashes(data.getBaseKey()));
+						// Set the hash to null to recompute it
+						backup.setDataHash(null);
+					}
+				}
```

Comment:
Existing external S3 Storage Locations will be modified to NOT end with one (or more) trailing slashes.